content/bash-scripting-basics.md

```diff
@@ -0,0 +1,158 @@
+
+The second crucial detail for your OS to properly run your script is to make it executable. As of right now, your script is just a regular text file. In order for your OS to run your script, we will need to make this text file executable. This can be done by navigating to our `~/scripts/` directory with our File Manager. Once you locate your script file, right click on the file to bring up with context menu and select "Properties". This will bring up the various details about the file we are looking at. Click on the tab labelled "Permissions". There will be a checkbox labelled "Allow executing file as program". Make sure to check this box. Once this is enabled, the script will be able to run.
+
+At the moment, the script will do nothing if we decided to execute it, so let's add some commands and then test our script. Open the script and add the following line below your first `shebang` line:
+
+```bash
+echo "The script executed correctly."
+```
+
+We can now attempt to run our script. Open the terminal from your Application Launcher. I am going to assume that you have followed along and created a directory in your Home directory called `scripts` and that you have named this script `basics.sh` (if you have placed it in a separate directory or named it differently, the steps are the same, but you must substitute the file names and paths). First, we need to navigate in our terminal to our `scripts` directory. Input the following command in your terminal to change directory to `~/scripts`:
+
+```bash
+cd ~/scripts
+```
+
+You can now run your script by running the following command in your terminal:
+
+```bash
+./basics.sh
+```
+
+If it worked, your terminal should output the following line:
+
+`The script executed correctly.`
+
+Congratulations! You have created a working Bash script. Of course, outputting a single line does not provide much functionality, so we will be doing just that in the next section.
+
+## Adding functionality
+
+At this point, our script does one thing: It outputs a single line in our terminal. The script works, but it isn't very useful yet. There are many things you can do with a script, such as if you combine several commands together to create something useful. There is a lot to cover so I won't be able to cover everything. If you would like to explore more ideas, a good place to find useful commands you can use in your script is by searching for what you need to accomplish on website such as [Stack Overflow](https://stackoverflow.com) or [AskUbuntu](https://askubuntu.com), and then after finding the appropriate command, add it to a script. You can also run each command on their own in a terminal to test what the command and it's options do.
```

Comment:
>There is a lot to cover so I won't be able to cover everything.

This feels like fluff now. Let's just remove it.

Reply:
>by searching for what you need to accomplish on website such as

"website" should be plural. That's also a doozy of a sentence. If you can trim it down or break it apart, that would probably be best.